addons/web_editor/static/tests/field_html_tests.js

```diff
@@ -348,6 +355,30 @@ QUnit.module('web_editor', {}, function () {
                 '<p>t<font style="background-color: rgb(0, 255, 255);">oto t</font><font style="" class="bg-o-color-3">oto to</font>to</p><p>tata</p>',
                 "should have rendered the field correctly in edit");
 
+            $('.note-editable').html("<p>ABCDEFG</p>");
+            var pText2 = $('.note-editable p')[0].firstChild;
+            Wysiwyg.setRange(pText2, 0, pText2, 3);
+            
+            await openColorpicker('#toolbar .note-fore-color-preview');
+            await testUtils.dom.click($('#toolbar .note-fore-color-preview .o_we_color_btn.bg-o-color-3'));
```

Comment:
To test the exact situation we have, which is the issue about hover. I need to perform multiple selections to simulate our scenario. It seems to work when I log the html, why is this an issue ?

Reply:
The issue is that you are adding to an existing test that is already clicking on a color, and with your changes the test looks like this:
```
clickOnColor(); // The old one.
resetContent();
clickOnColor(); // The new one.
hoverOnColor();
```
It's like it's two different tests smashed together because there is a reset of the content and then we do the same operation that was already there before. Two solutions:
A. you add to the existing test
```js
clickOnColor(); // The old one.
hoverOnColor();
```
B. you create your own test
```js
initializeContent();
clickOnColor();
hoverOnColor(); // The new one.
```

Reply:
I rewrote the test so as not to re-color on a new selection before testing.
At the end of the previous test, we end up with `<p>t<font>[oto toto ]</font>toto</p>`. Now we do the flow of what we're trying to fix: select the whole paragraph, hover a bunch of times on the colors, and check that the selection is still on the whole paragraph and not back to the contents of the font.